src/components/Admin/Onboard/ProductAddModal.vue

```diff
@@ -159,43 +163,64 @@
         }
         this.create_product_modal_view = true
       },
-      add_product: function () {
-        this.productAddedToast()
-        this.create_product_modal_view = true
-        // ajax call to verify and queue storing
-        this.current_category.products.push({
+      add_ready_product: function () {
+        shop.categoryProductCreate({
           image: this.new_product.image,
-          title: this.new_product.title,
           asset_id: this.new_product.asset_id,
+          store_id: this.current_category.store_id,
+          price_cents: this.new_product.price_cents * 100,
+          category_id: this.current_category.category_id,
           description: this.new_product.description,
-          price_cents: this.new_product.price_cents * 100
+          title: this.new_product.title,
+          display_price: this.new_product.price_cents
+        }).then(response => {
+          this.productAddedToast()
+          this.current_category.products.push(response.data)
+          this.reset_temp_product()
+        }).catch(error => {
+          setTimeout(Alert.create({
+            html: error.response.data.message,
+            color: 'red-7'
+          }).dismiss, 5000)
         })
-        if (this.$route.path === '/admin/products') {
-          shop.categoryProductCreate({
-            image: this.new_product.image,
-            asset_id: this.new_product.asset_id,
-            enabled: true,
-            store_id: this.current_category.store_id,
-            price_cents: this.new_product.price_cents * 100,
-            category_id: this.current_category.category_id,
-            description: this.new_product.description,
-            title: this.new_product.title,
-            display_price: this.new_product.price_cents
+      },
+      add_product: function () {
+        this.create_product_modal_view = true
+        // ajax call to verify and queue storing
+        if (!this.new_product.title || !this.new_product.price_cents) {
+          setTimeout(Alert.create({
```

Comment:
Writing this out every time gets annoying. Plus we might want to use something else one day, or the response pattern changes, etc. That would mean updating all these spots in the code. My apologies for spreading more copies of it lol. It should be easy to pull this out into a reusable/extensible method with sensible defaults. I can add something if you like